drivers/gpio/gpio_nrfx.c

```diff
@@ -0,0 +1,455 @@
+
+		pin &= ~BIT(cur_pin);
+
+		data->int_en &= ~BIT(cur_pin);
+
+		res = gpiote_pin_int_cfg(port, cur_pin);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	return res;
+}
+
+static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
+	.config = gpio_nrfx_config,
+	.write = gpio_nrfx_write,
+	.read = gpio_nrfx_read,
+	.manage_callback = gpio_nrfx_manage_callback,
+	.enable_callback = gpio_nrfx_enable_callback,
+	.disable_callback = gpio_nrfx_disable_callback
+};
+
+static u32_t get_trigger_pins(struct device *port)
+{
+	if (port == NULL) {
+		return 0;
+	}
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	u32_t port_in = nrf_gpio_port_in_read(get_port_cfg(port)->port);
+	u32_t xor = (port_in ^ data->active_level ^ data->polarity);
+	u32_t out = ~xor & data->int_en & data->pin_int_en
+		    & ~data->trig_edge & ~data->double_edge;
+
+	return out;
+}
+
+#ifdef CONFIG_GPIO_NRFX_P0
+DEVICE_DECLARE(gpio_nrfx_p0);
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+DEVICE_DECLARE(gpio_nrfx_p1);
+#endif
+
+static struct device *port_list[GPIO_COUNT] = {
+
+#ifdef CONFIG_GPIO_NRFX_P0
+	DEVICE_GET(gpio_nrfx_p0),
+#else
+	NULL,
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+	DEVICE_GET(gpio_nrfx_p1),
+#else
+	NULL,
+#endif
+};
+
+static void gpiote_event_handler(void)
+{
+	u32_t pins[GPIO_COUNT];
+	bool clear_port_evt = true;
+
+	memset(pins, 0, sizeof(pins));
+
+	if (nrf_gpiote_event_is_set(NRF_GPIOTE_EVENTS_PORT)) {
+		for (size_t i = 0; i < GPIO_COUNT; i++) {
+			pins[i] |= get_trigger_pins(port_list[i]);
+		}
+	}
+
+	for (size_t i = 0; i < GPIO_COUNT; i++) {
+		if (pins[i] != 0) {
+			clear_port_evt = false;
+			break;
+		}
+	}
+	if (clear_port_evt) {
+		nrf_gpiote_event_clear(NRF_GPIOTE_EVENTS_PORT);
+	}
+
+	for (size_t i = 0; i < GPIOTE_CH_NUM; i++) {
+		nrf_gpiote_events_t evt =
+			offsetof(NRF_GPIOTE_Type, EVENTS_IN[i]);
+
+		if (nrf_gpiote_event_is_set(evt)) {
+			u32_t pin = nrf_gpiote_event_pin_get(i);
+			u8_t port = ((pin & BIT(5)) >> 5);
+
+			pin &= BIT_MASK(5);
+			pins[port] |= BIT(pin);
+			nrf_gpiote_event_clear(evt);
+		}
+	}
+
+
+	for (size_t i = 0; i < GPIO_COUNT; i++) {
+		if (port_list[i] != NULL) {
+			_gpio_fire_callbacks(
+				&get_port_data(port_list[i])->callbacks,
+				port_list[i], pins[i]);
+		}
+	}
+}
+
+static int gpio_nrfx_init(struct device *port)
+{
+	static bool gpio_initialized;
+
+	if (!gpio_initialized) {
+		gpio_initialized = true;
+		IRQ_CONNECT(CONFIG_GPIOTE_NRF5_IRQ, CONFIG_GPIOTE_NRF5_IRQ_PRI,
+			    gpiote_event_handler, NULL, 0);
+
+		irq_enable(CONFIG_GPIOTE_NRF5_IRQ);
+		nrf_gpiote_int_enable(BIT(31));
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_SOC_SERIES_NRF51X
+#define NRF_P0 NRF_GPIO
+#endif
+
+#define GPIO_NRFX_DEVICE(id)						\
+static const struct gpio_nrfx_cfg gpio_nrfx_p##id##_cfg = {		\
+	.port = NRF_P##id,						\
+	.port_num = id							\
+};									\
+									\
+static struct gpio_nrfx_data gpio_nrfx_p##id##_data;			\
+									\
+DEVICE_AND_API_INIT(gpio_nrfx_p##id, CONFIG_GPIO_NRF5_P##id##_DEV_NAME,	\
```

Comment:
Here you are still using the "NRF5" nomenclature, which should not be done for the nrfx driver.
I understand that this might need some more work on the fixup/dts.